pandas_datareader/moex.py

```diff
@@ -60,17 +60,16 @@ def __init__(self, *args, **kwargs):
     def url(self):
         """Return a list of API URLs per symbol"""
 
-        if not self.__engines or not self.__markets:
+        if not self.__markets_n_engines:
             raise Exception(
                 "Accessing url property before invocation "
                 "of read() or _get_metadata() methods"
             )
 
         return [
             self.__url_data.format(
-                engine=self.__engines[s], market=self.__markets[s], symbol=s
-            )
-            for s in self.symbols
+                engine=engine, market=market, symbol=s
+            ) for s in self.symbols if s in self.__markets_n_engines for market, engine in self.__markets_n_engines[s]
```

Comment:
This line is too long. Probably avoid the comprehension.

Reply:
I split out a comprehension to next line. Comprehension is cosily.